### PR TITLE
[luci] Support channel-wise INT16 quantization

### PR DESCRIPTION
**Pull request by jinevening**
This supports channel-wise INT16 quantization

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: #4444 
Draft PR: #4445

### CODE REVIEW

File: compiler/luci/pass/src/PropagateConcatenationQparam.test.cpp [hunk 1]
```diff
@@ -48,19 +48,28 @@ int32_t quantize(float f, luci::CircleQuantParam *qparam)
...
+      addQuantParam(concat_node, {3.14}, {77});
+      addQuantParam(input_1, {1.0}, {1});
+      addQuantParam(input_2, {2.0}, {2});
+    }
+    else if (quant_type == loco::DataType::S16)
+    {
+      addQuantParam(concat_node, {3.14}, {0});
+      addQuantParam(input_1, {1.0}, {0});
+      addQuantParam(input_2, {2.0}, {0});
+    }
```
Comment (seanshpark) [inline compiler/luci/pass/src/PropagateConcatenationQparam.test.cpp:72]: maybe throw when not either of two?

Comment (jinevening) [reply compiler/luci/pass/src/PropagateConcatenationQparam.test.cpp:72]: I've added exceptions. Thanks.

File: compiler/luci/pass/src/PropagateConcatenationQparam.test.cpp [hunk 2]
```diff
@@ -48,19 +48,32 @@ int32_t quantize(float f, luci::CircleQuantParam *qparam)
...
+    else
+    {
+      throw std::runtime_error("Wrong quantization type");
```
Comment (seanshpark) [inline compiler/luci/pass/src/PropagateConcatenationQparam.test.cpp:75]: How about using `Invalid` or `Unsupported` instead of `Wrong`?

Comment (jinevening) [reply compiler/luci/pass/src/PropagateConcatenationQparam.test.cpp]: Updated

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

File: compiler/luci/pass/src/PropagateConcatenationQparam.test.cpp [hunk 3]
```diff
@@ -237,3 +275,98 @@ TEST(PropagateConcatenationQparam, propagate_concat_quantparam_NEG)
...
+  // (2) input used by other Op: input_1 is an input of input_2. qparam is propagated only to
+  // input_2
+  // (3) subsequent concat: input_1 is concat. qparam is propagated only to input_2
+  // (4) const input: input_1 is const. constant values are quantized
+
+  // normal case: qparam of concat_node is propagated to input_1 and input_2
+  SimpleConcatGraph g(loco::DataType::S16);
+  luci::propagate_concat_quantparam(&g.concat_node, loco::DataType::S16);
+  EXPECT_FLOAT_EQ(g.concat_node.quantparam()->scale[0], 3.14);
+  EXPECT_EQ(g.concat_node.quantparam()->zerop[0], 0);
```
Comment (seanshpark) [inline compiler/luci/pass/src/PropagateConcatenationQparam.test.cpp:292]: @mhs4670go proposed #256 to use like 
```cxx
EXPECT_EQ(0, g.concat_node.quantparam()->zerop[0]);
```
Can you please revise these ?

Comment (jinevening) [reply compiler/luci/pass/src/PropagateConcatenationQparam.test.cpp]: I didn't know the issue.. I revised them. Thanks.

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**